server/src/main/java/org/elasticsearch/index/snapshots/IndexShardSnapshotStatus.java

```diff
@@ -98,6 +98,7 @@ public enum AbortStatus {
     private long processedSize;
     private String failure;
     private final SubscribableListener<AbortStatus> abortListeners = new SubscribableListener<>();
+    private String debugStatusString;
```

Comment:
Probably needs to be `volatile` so that `toString()` and friends all see the latest value. And in that case there's no need to make the update method `synchronized`.

```suggestion
    private volatile String debugStatusString;
```